### ISSUE TITLE
Issue while installing node-rfc

### ISSUE THREAD

**Issue by Uday-Kiran** [SAP/node-rfc#302]
@bsrdjan kindly check help here.
**Describe the bug**
Not able to install node-rfc after all dependencies are made.

**To Reproduce**
Getting the same error when node-rfc is installed locally or globally.

**Screenshots**
kindly find attached verbose installation logs.
[2023-10-03T05_02_30_177Z-debug-0.log](https://github.com/SAP/node-rfc/files/12788858/2023-10-03T05_02_30_177Z-debug-0.log)

**Environment**
 - The output of `node -p "require('node-rfc').environment"`
<img width="609" alt="image" src="https://github.com/SAP/node-rfc/assets/15310022/e556891c-8c0a-49bd-b147-794d75eb6a44">

**Comment by bsrdjan** [comment]
The installation log shows that node-rfc binary build from source is triggered during installation, instead of using pre-built node-rfc fetched from NPM. The build from source should not be triggered on Windows and should work when requested by `npm install node-rtc --build-from-source`. In your installation the build from source is triggered for some reason and it fails.

I could not reproduce on Windows with node 20.7.0 and npm 10.1.0, both the standard `npm install node-rfc` and build from source installation worked.

Which npm version is installed on your system and what kind of system are you using? Standard Windows notebook, server, VM ...? Is there anything special with your Windows system or NPM configuration? What is the output of `npm config list` ?

I also see log lines not related to node-rfc, like express etc. Do you know why they show up? 

Can you do one more test in empty project directory like below and attach the installation log?

```shell
mkdir test
cd test
npm init -y
npm install --loglevel verbose node-rfc
```

**Comment by Uday-Kiran** [comment]
Hi @bsrdjan ,

I'm using standard windows laptop with Windows 11 enterprise OS.
Kindly check the output of `npm config list`
<img width="680" alt="Untitled" src="https://github.com/SAP/node-rfc/assets/15310022/dc39819e-fea5-4676-96cc-e43cd5b7a9e2">

please find attachment of installation logs.

**Comment by bsrdjan** [comment]
Did you test in new created empty directory? The first error is `86 warn cleanup     [Error: EPERM: operation not permitted`, like your user does not have permissions for certain operations.

Can you repeat the test with admin rights, like open the command console as Administrator and re-test ?

More convenient could be to install [NodeJS version manager](https://github.com/coreybutler/nvm-windows) and use it to install and activate the node version you want to use. Then no admin rights are needed:

```shell
nvm install 20.7.0
nvm use 20.7.0
mkdir test
npm init -y
npm install --loglevel verbose node-rfc
```

**Comment by Uday-Kiran** [comment]
Hi @bsrdjan ,

I repeated the steps by running command prompt as administrator and using node 20.7.0 using nvm.
Please find the attached logs.
[2023-10-09T13_28_05_294Z-debug-0.log](https://github.com/SAP/node-rfc/files/12846743/2023-10-09T13_28_05_294Z-debug-0.log)

**Comment by bsrdjan** [comment]
Hello @Uday-Kiran,

I see only one log. Is it the one with admin rights or with nvm. Can you send also the other one?

Are all pre-requisites installed and which version of SAP NW RFC SDK are you using? Need the output of

```shell
cd %SAPNWRFC_HOME%\bin
startrfc -v
```

Can you please also copy and send the CMD history of

```shell
mkdir issue302
cd issue302
npm init -y
npm install --loglevel verbose node-rfc
```

**Comment by Uday-Kiran** [comment]
Hi @bsrdjan ,

I followed installation instructions to install sap nw rfc sdk
![image](https://github.com/SAP/node-rfc/assets/15310022/5c3fa117-e295-4ea3-9644-a3abe9fce18a)

But not able to run startrfc
<img width="434" alt="image" src="https://github.com/SAP/node-rfc/assets/15310022/500b2f18-5437-4a69-a2fd-d158ddda42f7">

The above logs was generated by running cmd as administrator **and** nvm use 20.7.0 followed by
`mkdir test
npm init -y
npm install --loglevel verbose node-rfc`
Find below CMD history and corresponding logs
<img width="704" alt="image" src="https://github.com/SAP/node-rfc/assets/15310022/2d237d8e-7414-4203-823d-2a4fc1c7ab13">
[2023-10-09T17_18_40_531Z-debug-0.log](https://github.com/SAP/node-rfc/files/12848761/2023-10-09T17_18_40_531Z-debug-0.log)

**Comment by bsrdjan** [comment]
But is startrfc.exe file there? What is the output of dir command in %SAPNWRFC_HOME%\bin folder?

Users sometimes download SAP NWRFC SDK for the wrong platform. Did you check if correct RFC SDK is copied on your Windows system?

Here are [installation instructions](https://github.com/SAP/node-rfc/blob/main/doc/installation.md#sap-nwrfc-sdk-installation). Could you please verify and re-test startrfc ?

**Comment by Uday-Kiran** [comment]
Yes I followed the instructions to install SAP NWRFC SDK. Followed all steps to create environment variables.

I have downloaded the right files for windows. 
[nwrfc750P_12-70002726.zip](https://github.com/SAP/node-rfc/files/12865980/nwrfc750P_12-70002726.zip)

Check below directory:
<img width="743" alt="image" src="https://github.com/SAP/node-rfc/assets/15310022/32aba8d5-805b-4824-882a-5bb919a8eb94">

**Comment by bsrdjan** [comment]
As you can see the startrfc does not have `.exe` suffix, which means the SAP NW RFC SDK on your system is not for Windows platform. Many other users make unfortunately the same mistake. When the open the Service Portal page to download the SDK, users expect the SDK for their platform is automatically pre-selected and just download the package. It is unfortunately not the case and there is a dropdown box to select for which platform the SDK shall be downloaded.

Can you install the RFC SDK for Windows and re-test?

**Comment by Uday-Kiran** [comment]
Thanks @bsrdjan ,downloading the proper SDK for windows 64 bit resolved my issue.
Yes people can get confused and download the SDK on the download portal, as you said we have to select the required platform from dropdown and download necessary package.
